test/testsuites/operatorhubtests.go

```diff
@@ -90,18 +90,12 @@ func testDisableAllEnableOne(t *testing.T) {
 	err = toggle(t, 1, false, true)
 	require.NoError(t, err, "Error updating cluster OperatorHub")
 
+	err = checkCreated(1, namespace)
+	assert.NoError(t, err, "Default CatalogSource was not created")
+
 	err = checkCatsrcIsPresent(helpers.DefaultSources[0].Name, namespace)
 	assert.NoError(t, err, "Expected default CatalogSource is not present")
 
-	err = checkCatsrcIsDeleted(helpers.DefaultSources[1].Name, namespace)
```

Comment:
checkDeleted was checking only for the first n default CatalogSources, so I've added a new function that manages to check for default behavior on remaining CatalogSources

Reply:
@ankitathomas  My only issue with that is, what if the intention is: let's disable the 3rd source? This function won't be of much use since it's checking the `first` nr source. 

Here's a thought: why don't we keep everything as it was before, expect for one thing: inside `checkCatsrcIsDeleted` and `checkCatsrcIsCreated`, we poll for the resource's presence/absent instead of querying for it once and returning immediately? 

That way this test remains readable, and we make any calling site for these functions more robust.